awscli/customizations/s3/subcommands.py

```diff
@@ -533,6 +534,9 @@ def _display_page(self, response_data, use_basename=True):
                 filename = filename_components[-1]
             else:
                 filename = content['Key']
+            if strip_prefix:
+                if filename.startswith(strip_prefix):
```

Comment:
Unfortunately I do not think it will be that simple. Specifically, if a user provides the following command:
```
$ aws s3 ls s3://mybucket/a --recurisve
```
It should print things like:
```
abc
aws-cli/__init__.py
```

By solely stripping out the prefix, we will start getting this instead:
```
bc
ws-cli/__init__.py
```

I tried fixing this a long time ago: https://github.com/aws/aws-cli/pull/1009. It may be worth looking at the edge cases I ran into for reference. Looks like I at least put a good amount of comments for reference.